service/history/decision/handler_test.go

```diff
@@ -560,6 +565,169 @@ func TestHandleDecisionTaskStarted(t *testing.T) {
 	}
 }
 
+func TestHandleDecisionTaskCompleted(t *testing.T) {
+	_serializedTestToken := []byte("test-token")
+	_testTaskListName := "some-tasklist-name"
+	_testWorkflowTypeName := "some-workflow-type-name"
+	tests := []struct {
+		name               string
+		domainID           string
+		expectedErr        error
+		expectMockCalls    func(ctrl *gomock.Controller, decisionHandler *handlerImpl)
+		assertResponseBody func(t *testing.T, resp *types.HistoryRespondDecisionTaskCompletedResponse)
+		mutableState       *persistence.WorkflowMutableState
+	}{
+		{
+			name:        "failure to get domain from ID",
+			domainID:    _testInvalidDomainUUID,
+			expectedErr: &types.BadRequestError{Message: "Invalid domain UUID."},
+		},
+		{
+			name:        "token deserialazation failure",
+			domainID:    constants.TestDomainID,
+			expectedErr: workflow.ErrDeserializingToken,
+			expectMockCalls: func(ctrl *gomock.Controller, decisionHandler *handlerImpl) {
+				decisionHandler.tokenSerializer.(*common.MockTaskTokenSerializer).EXPECT().Deserialize(_serializedTestToken).Return(nil, errors.New("unable to deserialize task token"))
+			},
+		},
+		{
+			name:        "get or create wf execution failure",
+			domainID:    constants.TestDomainID,
+			expectedErr: &types.BadRequestError{Message: "Can't load workflow execution.  WorkflowId not set."},
+			expectMockCalls: func(ctrl *gomock.Controller, decisionHandler *handlerImpl) {
+				_taskToken := &common.TaskToken{
```

Comment:
We don't use _variable naming. Let's stick with regular naming patterns to avoid confusion. When I saw this I looked around to see whether this is a member variable or package level etc.

Reply:
same comment applies to all underscore '_' prefixed variable names in this file